### ISSUE TITLE
CLI serve action is not watching for changes

### ISSUE THREAD

**Issue by ronalson** [pattern-lab/patternlab-node-cli#54]
### General information

- NodeJS version: v8.9.1
- NPM version: 5.5.1

### Problem description

Pattern Lab CLI isn't properly watching and rebuilding patterns after changes are made.

### Reproduction steps

Install CLI

```bash
$ npm install pattern-lab/patternlab-node-cli --save-dev
```

Create npm tasks to use the CLI actions

```json
"pl:init": "patternlab init",
"pl:build": "patternlab build --config ./patternlab/patternlab-config.json",
"pl:serve": "patternlab serve --config ./patternlab/patternlab-config.json",
```
Init new PL project

```bash
$ npm run pl:init
? Please specify a directory for your PatternLab project. patternlab
? Which edition do you want to use (defaults to edition-node)? edition-node-gulp
? Which starterkit do you want to use? starterkit-mustache-demo
? Are you happy with your choices? (Hit enter for YES)? Yes

$ npm run pl:serve
...
Ready for changes
Serving "C:\...\my-project\patternlab\public" at http://127.0.0.1:8080
GET / 404 8.031 ms - 139
```
> **First Issue**
> Here the `server` task does not run the `build` task before launching server.

Next step:

```bash
$ npm run pl:build
```
PL build task goes through and generate everything inside public.

```bash
$ npm run pl:serve
```

Here, PL `serve` task starts serving the patterns at 8080. Everything's looks fine. However, there is no watching happening upon any file changes (source/css/style.css or source/_patterns etc). This is also the behavior even if I add the `--watch` flag to pl:serve or pl:build

---

Additional info:

`pl:init` as shown above installs these dependencies to the project:

```json
{
...
 "dependencies": {
    "@pattern-lab/starterkit-mustache-demo": "^4.0.4",
    "edition-node-gulp": "^1.4.0"
  }
}
```

**Comment by raphaelokon** [comment]
Watches work for me. But I agree that there should be a build before the initial serve.

**Comment by ronalson** [comment]
Are you passing the `--watch` flag? (This is not clear for me from the readme).

I did all tests above on a windows machine. I'll try again at home on my macbook.

**Comment by raphaelokon** [comment]
@ronalson Aye. You need to pass the `--watch` flag in terms for rebuilding.

Also: I implemented the initial build on `serve` btw so you won't see the 404 once this lands. Pushing this over the weekend.

**Comment by ronalson** [comment]
Hey @raphaelokon, I would link to thank you for the hard work and quick response on this issue. However, I was not able to use the CLI without running into the issue above.

On my windows machine, the watching is not happening, even if I use the --watch flag as below.
```
"pl:serve": "patternlab serve --watch --config ./patternlab/patternlab-config.json"
```
Am I using it right?

---

On my OSX machine, I tried many times this weekend but I could not make it work either. I was able to run `npm run pl:init` (which fires patternlab init), but after that, while running build or serve I got some node ELIFECYCLE errors. One thing I notice was that after the `patternlab init` task, the **patternlab-node-cli** disappeared from within node_modules.

This might be some weird config of my side. I'll investigate further, but I just want to give you a feedback. I took some prints, but I don't have them with me. If you fill it would be helpful, let me know and I will post it here later at night.

Thanks.
